### PR TITLE
Added the ability for plot to put the cube data on the x axis if plotting just a cube against a vertical or y coordinate.

### LINKED ISSUE #612: Automatic selection of vertical axis in 1D plots
Since #593 it is possible to make 1D plots with the coordinate on the y-axis and cube data on the x-axis. It is proposed that when a call to a 1D plotting function is made in the form:

```
plot(cube)
```

and when the cube's dimension coordinate is a vertical coordinate, the plot should automatically have the coordinate on the y-axis.

The rationale for this is that iris should always try and produce the most sensible plot when given no specific instructions about coordinates.

### PR DESCRIPTION
**Pull request by pelson**
Closes #612.

Would need a what's new.

### CODE REVIEW

**Comment by pelson** [comment]
Looks like a genuine failure. Old test: https://scitools.github.io/test-iris-imagehash/images/f15f832a5ee0492a36e8b9ed8fa4f2b629dace4921681e17a9807e7c89835ef0.png

New test:

result-iris tests test_plot test1dplotmultiargs test_cube 0

**Comment by ajdawson** [comment]
That is an altltude vs temperature plot, the new one is correct for the change you made.

**Comment by ajdawson** [comment]
Others are either altitude vs temperature or latitude vs temperature, both of which should be affected by this PR, so all looks good to me.

File: lib/iris/plot.py [hunk 1]
```diff
@@ -479,7 +479,19 @@ def _get_plot_objects(args):
...
         v_object = args[0]
         u_object = _u_object_from_v_object(v_object)
+
         u, v = _uv_from_u_object_v_object(u_object, args[0])
+
+        # If a single cube argument, and the associated dimension coordinate
+        # is vertical-like, put the coordinate on the y axis, and the data o
+        # the x.
+        if (isinstance(v_object, iris.cube.Cube)
+                and iris.util.guess_coord_axis(u_object) in ['Y', 'Z']):
```
Comment (ajdawson) [inline lib/iris/plot.py]: This is your coding standards error, the and should be on the previous line.

File: lib/iris/quickplot.py [hunk 2]
```diff
@@ -127,11 +127,12 @@ def _get_titles(u_object, v_object):
 
 
 def _label_1d_plot(*args):
-    if len(args) > 1 and isinstance(args[1],
+    u_obj, v_obj, _, _, _ = iplt._get_plot_objects(args)
+    if len(args) > 1 and isinstance(u_obj,
```
Comment (pelson) [inline lib/iris/quickplot.py]: If statement is now not needed.

**Comment by ajdawson** [comment]
Probably a small amount of effort needed to bring this up to speed.

**Comment by pelson** [comment]
I've rebased and fixed the pep8 issue. I'm unlikely to get the time to do the image hashes soon unfortunately.

**Comment by pelson** [comment]
Closing as this has gone stale. Implementation is solid if somebody want to take it on and get it over the finishing line.

**Pull request closed without merging**